fdbserver/workloads/FastTriggeredWatches.actor.cpp

```diff
@@ -144,9 +148,10 @@ struct FastTriggeredWatchesWorkload : TestWorkload {
 				Version ver = wait(setFuture);
 				//TraceEvent("FTWWatchDone").detail("Key", printable(setKey));
 				// Assert that the time from setting the key to triggering the watch is no greater than 25s
-				// TODO: This assertion can cause flaky behaviour since sometimes a watch can take longer to fire
-				ASSERT(first || lastReadVersion - ver >= SERVER_KNOBS->MAX_VERSIONS_IN_FLIGHT ||
-				       lastReadVersion - ver < SERVER_KNOBS->VERSIONS_PER_SECOND * (25 + getDuration));
+				ASSERT(!watchCommitted ||
+				       lastReadVersion - std::max(ver, watchCommitVersion) >= SERVER_KNOBS->MAX_VERSIONS_IN_FLIGHT ||
```

Comment:
Nit: can we rename `ver` for clarity, since we now have three versions being evaluated?

Reply:
Done